### PR TITLE
Feature/optifine

### PR DESCRIPTION
**Pull request by marchermans**
This enables the use of shaders when using Structurize together with Optifine.

Most features are supported, but certain things will produce visual artifacts, like hard shadows on the sides turned away from the sun. There is nothing i can do in that case due to me not being able to disable this from code, but only from the shader, which i have no control over.

Again thanks go to: @sp614x for the help here.

### CODE REVIEW

File: src/main/java/com/structurize/compat/optifine/OptifineCompat.java [hunk 1]
```diff
@@ -0,0 +1,188 @@
+package com.structurize.compat.optifine;
```
Comment (Raycoms) [inline src/main/java/com/structurize/compat/optifine/OptifineCompat.java:1]: Can you please run our formatter on this one?

File: build.gradle [hunk 2]
```diff
@@ -29,6 +29,9 @@ repositories {
     maven {
         url 'http://dogforce-games.com/maven'
     }
+    flatDir {
```
Comment (Raycoms) [inline build.gradle]: What does this do? Do we need this?

File: src/main/java/com/structurize/compat/optifine/OptifineCompat.java (hunk 1 above)
Comment (marchermans) [reply src/main/java/com/structurize/compat/optifine/OptifineCompat.java:1]: Did I not do that?

Comment (Raycoms) [reply src/main/java/com/structurize/compat/optifine/OptifineCompat.java:1]: nope =D

**Pull request merged**